src/storage/PutProcessor.cpp

```diff
@@ -0,0 +1,29 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "storage/PutProcessor.h"
+
+namespace nebula {
+namespace storage {
+
+void PutProcessor::process(const cpp2::PutRequest& req) {
+    const auto& pairs = req.get_parts();
+    auto space = req.get_space_id();
+    callingNum_ = pairs.size();
+    CHECK_NOTNULL(kvstore_);
+
+    std::for_each(pairs.begin(), pairs.end(), [&](auto& value) {
+        auto part = value.first;
+        std::vector<kvstore::KV> data;
```

Comment:
Maybe we can collect all pairs and put only once

Reply:
Seems `doPut` would process each partition at one time.

I will use folly::collectAll().via().then() to process put in parallel. Similar with GetProcessor